### PR TITLE
DDP-5433 enforce required mailing address fields via SDK configuration flag

### PR DESCRIPTION
**Pull request by MocanaAtBroad**

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 1]
```diff
@@ -201,19 +205,20 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
...
-    });
-    this.initializeComponentState();
-  }
+  private readonly LOG_SOURCE = 'AddressEmbeddedComponent';
+
+    constructor(
+        private addressService: AddressService,
+        private logger: LoggingService,
+        private ngxTranslate: NGXTranslateService,
+        @Inject('ddp.config') private configuration: ConfigurationService,
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts:214]: Get the configuration

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 2]
```diff
@@ -69,6 +71,7 @@ interface AddressSuggestion {
         <p *ngIf="block.subtitleText" class="ddp-address-embedded__subtitle" [innerHTML]="block.subtitleText"></p>
         <ddp-address-input
                 (valueChanged)="inputComponentAddress$.next($event)"
+                (formValidStatusChanged)="formValidStatusChanged$.next($event)"
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts:74]: Getting the form status from the child component.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 3]
```diff
@@ -553,9 +558,11 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
...
 
-    const emitValidStatusAction$ = combineLatest([this.formErrorMessages$, this.verifyFieldErrors$, activitityRequirementsMet$]).pipe(
-        map(([formErrors, addressErrors, reqsMet]) =>
-            !formErrors.length && !addressErrors.length && reqsMet),
+    const emitValidStatusAction$ = combineLatest([this.formValidStatusChanged$, this.formErrorMessages$, this.verifyFieldErrors$,
+        activitityRequirementsMet$]).pipe(
+        map(([formValidStatusChanged, formErrors, addressErrors, reqsMet]) =>
+            (!this.configuration.addressEnforceRequiredFields || formValidStatusChanged)
+                && !formErrors.length && !addressErrors.length && reqsMet),
         distinctUntilChanged(),
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts:566]: We care about the valid status for overall component valid status if configuration flag is set to true.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressInput.component.ts [hunk 4]
```diff
@@ -237,15 +243,23 @@ export class AddressInputComponent implements OnInit, OnDestroy {
     const valueChangedEmit$ = this.ais.addressOutputStream$.pipe(
       tap(val => this.valueChanged.emit(val)),
     );
+
+    const formValidStatusChangedEmit$ = merge(this.ais.inputAddress$, this.ais.addressForm.valueChanges).pipe(
+          tap(val => this.formValidStatusChanged.emit(this.ais.addressForm.valid))
+    );
+
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressInput.component.ts:250]: Not enough to emit on addressForm.valueChanges . Sometimes we set the form values from parent component with the emit event=false flag, so we need to merge that event here as well.

**Review by fat23cat — CHANGES_REQUESTED** [review]
Changes look good. But could you please fix the tests? Some of them seem to be broken.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.spec.ts [hunk 5]
```diff
@@ -19,6 +19,7 @@ import { AddressVerificationResponse } from '../../models/addressVerificationRes
 import { NGXTranslateService } from '../../services/internationalization/ngxTranslate.service';
 import { TranslateTestingModule } from '../../testsupport/translateTestingModule';
 import { LoggingService } from '../../services/logging.service';
+import { ConfigurationService } from 'ddp-sdk';
```
Comment (fat23cat) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.spec.ts]:
```suggestion
import { ConfigurationService } from '../../services/configuration.service';
```

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.spec.ts [hunk 6]
```diff
@@ -56,13 +59,14 @@ class FakeAddressInputComponent {
   }
 }
 
-describe('AddressEmbeddedComponent', () => {
+fdescribe('AddressEmbeddedComponent', () => {
```
Comment (fat23cat) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.spec.ts]:
```suggestion
describe('AddressEmbeddedComponent', () => {
```

**Review by fat23cat — CHANGES_REQUESTED** [review]
Just a couple of suggestions.

**Review by fat23cat — APPROVED** [review]

**Pull request merged**